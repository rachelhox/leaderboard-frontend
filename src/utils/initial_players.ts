/* eslint-disable no-undef */
export const initialPlayers: Player[] = [
  {
    name: 'Angel',
    scores: 35,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Angel`,
  },
  {
    name: 'Angus',
    scores: 98,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Angus`,
  },
  {
    name: 'Berry',
    scores: 8,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Berry`,
  },
  {
    name: 'Henry',
    scores: 98,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Henry`,
  },
  {
    name: 'Louis',
    scores: 56,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Louis`,
  },
  {
    name: 'Gus',
    scores: 29,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Gus`,
  },
  {
    name: 'Hannah',
    scores: 79,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Hannah`,
  },
  {
    name: 'Manny',
    scores: 59,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Manny`,
  },
  {
    name: 'Catherine',
    scores: 36,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Catherine`,
  },
  {
    name: 'Audrey',
    scores: 48,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Audrey`,
  },
];
