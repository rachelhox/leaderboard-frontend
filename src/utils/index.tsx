import { initialPlayers } from './initial_players';
import { getComparator, stableSort } from './sort';

export { initialPlayers, getComparator, stableSort };
