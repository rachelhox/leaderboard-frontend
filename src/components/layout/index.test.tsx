import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Layout from '.';

// ==================================
// global setup
// ==================================
let component: renderer.ReactTestRenderer;

// ==================================
// mocks
// ==================================

jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/',
  }),
}));

// ==================================
// unit tests
// ==================================
describe('component: Layout', () => {
  beforeEach(() => {
    component = renderer.create(
      <MockTheme>
        <Layout>
          <div>hello world</div>
        </Layout>
      </MockTheme>
    );
  });

  it('it renders', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
