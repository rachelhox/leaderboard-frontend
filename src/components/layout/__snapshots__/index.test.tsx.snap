// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Layout it renders 1`] = `
<div
  className="MuiBox-root css-79elbk"
>
  <div
    className="MuiBox-root css-1408fvk"
  >
    <div
      className="MuiBox-root css-6y38p3"
    >
      <div>
        hello world
      </div>
    </div>
  </div>
</div>
`;
